### PR TITLE
feat: add waitForTx to the accept bid flow

### LINKED ISSUE #1093: Wait for the transaction to be completed when accepting a bid of a LAND with a rental listing
## 🎉 Description

Following this PR, when the user accepts a bid and there's an open rental, wait for the transaction to be completed to later wait for the rental listing to be cancelled.
    if (nft?.openRentalId) {
      yield call(waitForTx, txHash)
      const rental: RentalListing | null = yield select(
        getRentalById,
        nft.openRentalId
      )
      if (isRentalListingOpen(rental)) {
        yield call(waitUntilRentalChangesStatus, nft, RentalStatus.CANCELLED)
      }
    }
Remember to create a new action to send the transaction to the state before waiting for it and to change the track and Transaction files to use the new action.

* [ ] 🐛 This is a bug report.
* [x] 📈 This is a feature request.

### PR DESCRIPTION
**Pull request by juanmahidalgo**
Closes #1093

### CODE REVIEW

File: webapp/src/modules/bid/actions.ts [hunk 1]
```diff
@@ -84,9 +86,22 @@ export const acceptBidSuccess = (bid: Bid, txHash: string) =>
 export const acceptBidFailure = (bid: Bid, error: string) =>
   action(ACCEPT_BID_FAILURE, { bid, error })
 
+export const acceptBidtransactionSubmitted = (bid: Bid, txHash: string) =>
```
Comment (LautaroPetaccio) [inline webapp/src/modules/bid/actions.ts:89]: Would you mind removing the transaction data from `acceptBidSuccess`?

File: webapp/src/modules/analytics/track.ts [hunk 2]
```diff
@@ -153,7 +153,7 @@ track<PlaceBidSuccessAction>(
 )
 
 track<AcceptBidSuccessAction>(
-  ACCEPT_BID_SUCCESS,
+  ACCEPT_BID_TRANSACTION_SUBMITTED,
```
Comment (LautaroPetaccio) [inline webapp/src/modules/analytics/track.ts:156]: When tracking this action, the payload is built using bid, but `ACCEPT_BID_TRANSACTION_SUBMITTED` is not receiving a bid when it's created.

File: webapp/src/modules/bid/actions.ts [hunk 3]
```diff
@@ -84,9 +86,22 @@ export const acceptBidSuccess = (bid: Bid, txHash: string) =>
...
+  action(
+    ACCEPT_BID_TRANSACTION_SUBMITTED,
```
Comment (LautaroPetaccio) [inline webapp/src/modules/bid/actions.ts:91]: We should add bid here to use it later in the track file:
```suggestion
    ACCEPT_BID_TRANSACTION_SUBMITTED,
    bid,
```

File: webapp/src/modules/bid/sagas.spec.ts [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+    })
+  })
+
+  describe('when accepting a bid request fails', () => {
+    it('should dispatch an action signaling the failure of the action handling', () => {
+      const bid = {
+        contractAddress: '0x123'
+      } as Bid
+      const address = 'anAddress'
+      const wallet = { address } as Wallet
```
Comment (LautaroPetaccio) [inline webapp/src/modules/bid/sagas.spec.ts:80]: This is repeated in some of the tests, if you use a beforeEach on an upper statement, you can re-use them if you want.

**Review by LautaroPetaccio — CHANGES_REQUESTED** [review]
Beside these comments, the PR looks great! Thanks for adding the whole set of tests for the sagas, they look great!

File: webapp/src/modules/analytics/track.ts (hunk 2 above)
Comment (juanmahidalgo) [reply webapp/src/modules/analytics/track.ts:156]: ooh true, good catch! added!

File: webapp/src/modules/bid/sagas.spec.ts (hunk 4 above)
Comment (juanmahidalgo) [reply webapp/src/modules/bid/sagas.spec.ts]: for sure, updated!

File: webapp/src/modules/bid/actions.ts (hunk 1 above)
Comment (juanmahidalgo) [reply webapp/src/modules/bid/actions.ts:84]: sure, updated!

**Review by LautaroPetaccio — APPROVED** [review]

**Pull request merged**